packages/trader/src/App/Components/Elements/VideoStream/video-stream.tsx

```diff
@@ -1,75 +1,305 @@
+
+    const video_ref = React.useRef<StreamPlayerApi>();
+    const progress_bar_ref = React.useRef<HTMLDivElement>(null);
+    const animation_ref = React.useRef<number>(0);
+    const timer_ref = React.useRef<ReturnType<typeof setTimeout>>();
+    const is_dragging = React.useRef<boolean>(false);
+
+    const onVolumeChange = (new_value: number) => {
+        if (!video_ref?.current) return;
+
+        video_ref.current.volume = new_value;
+    };
+
+    const toggleMute = (new_value: boolean) => {
+        if (!video_ref?.current) return;
+
+        video_ref.current.muted = new_value;
+    };
+
+    const onPlaybackRateChange = (new_value: number) => {
+        if (!video_ref?.current) return;
+
+        video_ref.current.playbackRate = new_value;
+    };
+
+    const togglePlaying = (e: React.MouseEvent<HTMLElement, MouseEvent> | React.KeyboardEvent<HTMLElement>) => {
+        e.stopPropagation();
+
+        if (!video_ref?.current || !progress_bar_ref?.current) return;
+
+        if (is_playing) {
+            video_ref.current.pause();
+            progress_bar_ref.current.style.setProperty('transition', 'none');
+            cancelAnimationFrame(animation_ref.current);
+        } else {
+            video_ref.current.play();
+        }
+
+        setIsPlaying(prev => !prev);
+        setIsEnded(false);
+    };
+
+    const calculateNewWidth = (e: React.MouseEvent<HTMLElement, MouseEvent> | React.TouchEvent<HTMLElement>) => {
+        const progress_bar = document.querySelector('.player__controls__progress-bar');
+        const client_X =
+            e.type === 'mousemove' || e.type === 'click'
+                ? (e as React.MouseEvent<HTMLElement, MouseEvent>).clientX
+                : (e as React.TouchEvent<HTMLElement>).changedTouches[0].clientX;
+
+        let new_width =
+            ((client_X - shift_X - (progress_bar?.getBoundingClientRect().left ?? 0)) /
+                (progress_bar?.getBoundingClientRect().width ?? 0)) *
+            100;
+        if (new_width >= 100) {
+            new_width = 100;
+        }
+        if (new_width <= 0) {
+            new_width = 0;
+        }
+
+        return new_width;
+    };
+
+    const dragStartHandler = (e: React.MouseEvent<HTMLElement, MouseEvent> | React.TouchEvent<HTMLElement>) => {
+        if (e.type === 'mousedown') e.preventDefault();
+        e.stopPropagation();
+
+        const point = document.querySelector('.player__progress-dot') as HTMLElement;
+        const client_X =
+            e.type === 'mousedown'
+                ? (e as React.MouseEvent<HTMLElement, MouseEvent>).clientX
+                : (e as React.TouchEvent<HTMLElement>).changedTouches[0].clientX;
+        setShiftX(client_X - (point?.getBoundingClientRect().left ?? 0));
+
+        video_ref?.current?.pause();
+        setIsPlaying(false);
+
+        is_dragging.current = true;
+
+        if (is_mobile) {
+            point.style.width = '1.6rem';
```

Comment:
According to the tasks requirements, the size of the drag dot should increase on hover/dragging. On desktop I was able to do this with only with css (:hover), but on mobile I have to do it via js